pkg/actions/lua/storage/aws/glue.go

```diff
@@ -0,0 +1,224 @@
+		var tableInput types.TableInput
+		err := json.Unmarshal([]byte(tableInputJSON), &tableInput)
+		if err != nil {
+			lua.Errorf(l, err.Error())
+			panic("unreachable")
+		}
+
+		// check if catalog ID provided
+		var catalogID *string
+		if !l.IsNone(3) {
+			catalogID = aws.String(lua.CheckString(l, 3))
+		}
+
+		// version Id optional
+		var versionID *string
+		if !l.IsNone(4) {
+			versionID = aws.String(lua.CheckString(l, 4))
+		}
+
+		// glue skip-archive optional
+		var skipArchive *bool
+		if !l.IsNone(5) {
+			lua.CheckType(l, 5, lua.TypeBoolean)
+			skipArchive = aws.Bool(l.ToBoolean(5))
+		}
+
+		_, err = client.UpdateTable(c.ctx, &glue.UpdateTableInput{
+			DatabaseName: &database,
+			TableInput:   &tableInput,
+			CatalogId:    catalogID,
+			VersionId:    versionID,
+			SkipArchive:  skipArchive,
+		})
+
+		if err != nil {
+			lua.Errorf(l, err.Error())
+			panic("unreachable")
+		}
+		return 0
+	}
+}
+func createTable(c *GlueClient) lua.Function {
+	return func(l *lua.State) int {
+		client := c.client()
+		database := lua.CheckString(l, 1)
+		tableInputJSON := lua.CheckString(l, 2)
+		// parse table input JSON
+		var tableInput types.TableInput
+		err := json.Unmarshal([]byte(tableInputJSON), &tableInput)
+		if err != nil {
+			lua.Errorf(l, err.Error())
+			panic("unreachable")
+		}
+		// check if catalog ID provided
+		var catalogID *string
+		if !l.IsNone(3) {
+			catalogID = aws.String(lua.CheckString(l, 3))
+		}
+		// TODO(isan) Additional input params: partition index and iceberg table format
+		// AWS API call
+		_, err = client.CreateTable(c.ctx, &glue.CreateTableInput{
+			DatabaseName: aws.String(database),
+			TableInput:   &tableInput,
+			CatalogId:    catalogID,
+		})
+		if err != nil {
+			lua.Errorf(l, err.Error())
+			panic("unreachable")
+		}
+		return 0
+	}
+}
+
+func getTable(c *GlueClient) lua.Function {
+	return func(l *lua.State) int {
+		client := c.client()
+		database := lua.CheckString(l, 1)
+		table := lua.CheckString(l, 2)
+		var catalogID *string
+		if !l.IsNone(3) {
+			catalogID = aws.String(lua.CheckString(l, 3))
+		}
+		resp, err := client.GetTable(c.ctx, &glue.GetTableInput{
+			DatabaseName: aws.String(database),
+			Name:         aws.String(table),
+			CatalogId:    catalogID,
+		})
+		if err != nil {
+			var notFoundErr *types.EntityNotFoundException
+			if errors.As(err, &notFoundErr) {
+				l.PushString("")
+				l.PushBoolean(false) // exists
+				return 2
+			}
+			lua.Errorf(l, err.Error())
+			panic("unreachable")
+		}
+		// Marshal the GetTableOutput struct to JSON.
+		jsonBytes, err := json.Marshal(resp)
+		if err != nil {
+			lua.Errorf(l, err.Error())
+			panic("unreachable")
+		}
+		// Unmarshal the JSON to a map.
+		var itemMap map[string]interface{}
+		err = json.Unmarshal(jsonBytes, &itemMap)
+		if err != nil {
+			lua.Errorf(l, err.Error())
+			panic("unreachable")
+		}
+		retArgs := util.DeepPush(l, itemMap)
+		l.PushBoolean(true)
+		return retArgs + 1
```

Comment:
Strange that DeepPush needs to return a number - it expected to already return 1.
Same for this function `getTable` return - we must return 2 as we like the boolean to be on position 2.

Reply:
Good catch, changed 👍